app/helpers/argo_helper.rb

```diff
@@ -179,7 +179,7 @@ def render_foxml_link(document, link_text = 'FoXML', opts = {:target => '_blank'
   end
 
   def render_solr_link(document)
-    return '' unless current_user.is_admin
+    return '' unless can? :read, document
```

Comment:
If the can-read-doc is itself based on is_admin criteria, then no problem there! Just wanted to make sure it wasn’t trying to bring in more generic privileges to the item. Thanks.

> On Feb 26, 2016, at 2:22 PM, Chris Beer <notifications@github.com> wrote:
> 
> In app/helpers/argo_helper.rb:
> 
> > @@ -179,7 +179,7 @@ def render_foxml_link(document, link_text = 'FoXML', opts = {:target => '_blank'
> >    end
> >  
> >    def render_solr_link(document)
> > -    return '' unless current_user.is_admin
> > +    return '' unless can? :read, document
> 
> @LynnMcRae cancan provides the ability to separate the "who" from the "what". The view asks "who can read (view) this document", and the ability class says "admins can read (view) documents).
> 
> —
> Reply to this email directly or view it on GitHub.
>